libnetwork/netutils/utils_linux.go

```diff
@@ -104,7 +104,7 @@ func ElectInterfaceAddresses(name string) ([]*net.IPNet, []*net.IPNet, error) {
 
 // FindAvailableNetwork returns a network from the passed list which does not
 // overlap with existing interfaces in the system
-func FindAvailableNetwork(list []*net.IPNet) (*net.IPNet, error) {
+func FindAvailableNetwork(ipVer ipamutils.IPVersion, list []*net.IPNet) (*net.IPNet, error) {
```

Comment:
Alternative, we could have a filter function that can be passed (in this case filter on IP class)